### DISCUSSION TITLE
New or Old product when duplicating / cloning

### DISCUSSION THREAD

**Discussion by jokn74** [PrestaShop/PrestaShop#35640] [Ideas]
### The problem

When duplicating a product the "new" product has the old creation date copied, but in most cases the new product is a new product and not an old product.

### The solution

When duplicating a pop ud could ask if the duplicate should be considered a new product, with today as creation date.

### Alternatives

There should be to different duplicate options:
1: Duplicate as new product
2: Duplicate as old product

### Additional context

The database contains the date both in [ps_product] and [ps_product_shop] so both dates should be modified to the desired date.

### Do you plan to work on this subject?

- [ ] I'm willing to contribute a [formal specification](https://github.com/PrestaShop/prestashop-specs).
- [ ] I'm willing to provide any wireframes or design assets required for this feature.
- [X] I'm willing to submit a Pull Request that implements this feature.
- [X] I'm willing to help verify that the implemented feature works as intended and produces no unintended side effects.

**Comment by matthieu-rolland** [comment]
Totally agree about the problem.

`date_add` should only be set by the object model with the current date at the time of creation, not get the date from the old product.

I'm currently working on a problematic linked to this date_add field on product tables, I'll take care of this as well.

**Comment by jokn74** [comment]
Hi

I am on PrestaShop-version: 8.1.4

Med venlig hilsen
Jeppe
***@***.***

Fra: Matthieu Rolland ***@***.***>
Sendt: 14. marts 2024 11:27
Til: PrestaShop/PrestaShop ***@***.***>
Cc: Jeppe (KOK MC ApS) ***@***.***>; Mention ***@***.***>
Emne: Re: [PrestaShop/PrestaShop] New or Old product when duplicating / cloning (Discussion #35640)


@jokn74<https://github.com/jokn74> it seems like in the product page v2 the date_add field is not duplicated, it takes the date of duplication.

On what version of PrestaShop are you ?

**Reply by jolelievre** [reply]
yep my bad, I tested on new product page but checked the wrong column and gave wrong info to @matthieu-rolland 😅
That's why he deleted his message I think, but the bug is indeed on the new product page and should be fixed, good catch

**Reply by matthieu-rolland** [reply]
I have a fix, a pr will come soon 👍

The product is actually correctly duplicated with the current date in `date_add`, but it is overriden just after with former `date_add` value during the shop association. It doesn't seem voluntary but rather a bug. @jokn74

**Reply by kpodemski** [reply]
@jolelievre @matthieu-rolland https://github.com/PrestaShop/PrestaShop/pull/34823
